### PR TITLE
Quantum Self Supervised Learning and Quantum LLM

### PR DESCRIPTION
**Pull request by CassNot**
In this repo:

1. I am implementing a photonic version of the "Quantum Self Supervised Learning" work whose paper can be found [here](https://arxiv.org/abs/2103.14653) and code [here](https://github.com/bjader/QSSL).

The `main.py` file contains the necessary tools to:
- run the classical baseline following the original implementation
- run the Qiskit implementation from Jaderberg et al., more precisely, with the `--q_backend qasm_simulator`, using `--qiskit`
- run a simple MerLin implementation using `--merlin` (Note that the configuration of this QuantumLayer can be changed using `--modes`)

The ReadMe file explains how to run the model in these different settings and presents the results
A Notebook presents a simple implementation of this framework to emphasize the simplicity and speed of this framework.

_Note: I am not using the argument of the computational speed-up here_

2. I am implementing a photonic version of the "Quantum Large Language Model Fine Tuning" work whose paper can be found [here](https://arxiv.org/abs/2504.08732)

The `main.py` file contains the necessary tools to:
- run the classical baseline following the suggested implementation in the paper
- run the TorchQuantum implementation designed with information given in the paper
- run a 3 different MerLin implementation while changing the number of quantum modes
- learning_rate, epochs and batch sizes can also be changed as arguments
The ReadMe file explains how to run the model in these different setting
A Notebook presents the different frameworks with their results and some data analysis.

### CODE REVIEW

**Review by philippeschoeb — CHANGES_REQUESTED** [review]
The only comment i have concerns the qssl_notebook.ipynb:
The conclusion says that there is no clear advantage of one algorithm over the other, however I obtain that the best accuracy with MerLin is 0.67% and that the best one with the classical SSL approach is 0.77%, which is substantial.

**Comment by CassNot** [comment]
I updated the conclusion in the QSSL

**Comment by CassNot** [comment]
Thank you for all the reviews ! 

- When training models for several epochs from the command line (default hps), print the metrics for the last epoch (currently only prints for the first epoch and every 10th epoch). I suggest this because when training for 5 epochs, for examples, if only metrics from the first epoch are shown, it is not clear whether the model was trained for 5 epochs or not.
- [x] print the metrics for last epoch

- When training classical svm & log-reg models from the command line (default hps), print final metrics.
- [x] I removed the conditional print statements dependant on args.verbose, now, it should print all the time

- In the README, at the start, add a sentence or two to explain what exactly is the classification task.
- [x] I added a reference to the paper and precised the dataset 

- In the qLLM notebook, an import matplotlib.pyplot as plt is missing from the import block.
- [x] I moved this import at the top of the files with all the other importations
